src/vanilla.ts

```diff
@@ -241,21 +241,21 @@ export const subscribe = (
   }
 }
 
-export type DeepResolveType<T> = T extends (...args: any[]) => any
+export type Snapshot<T> = T extends (...args: any[]) => any
   ? T
   : T extends AsRef
-  ? T
+  ? Readonly<T>
   : T extends Promise<infer V>
-  ? V
+  ? Readonly<V>
   : T extends object
   ? {
-      [K in keyof T]: DeepResolveType<T[K]>
+      readonly [K in keyof T]: Snapshot<T[K]>
     }
-  : T
+  : T extends any
```

Comment:
How does this trick work? What was the issue without it?

Reply:
Originally a test broke that was using `proxy<any>` which lead to Typescript complaining about `Readonly<any>` not being valid or something like that. But I actually can't reproduce this anymore. Maybe my IDE or the TS server had gotten stuck. Anyway I'm removing the code again.